src/pages/dev/general-message-passing/gas-services/pay-gas.mdx

```diff
@@ -50,92 +50,47 @@ contract SimpleTransferContract {
-function payNativeGasForContractCall(
-    address sender,
-    string calldata destinationChain,
-    string calldata destinationAddress,
-    bytes calldata payload,
-    address refundAddress
-) external payable;
-```
-
-## Alternative gas payment methods for `callContractWithToken`
+After sending a transaction out, Axelar’s Executor Service will do the following:
 
+- Monitor `AxelarGasReceiver` for receipt of payment and get the amount paid as `amountPaid`.
+- Match the gas paid to the correct contract calls.
+- Execute the contract call with the `gasLimit` specified by the application.
 
-### payNativeGasForContractCallWithToken
-This method accepts the native tokens of the source chain.
+## Alternative gas payment methods
 
-```solidity
-// This is called on the source chain before calling the gateway to execute a remote contract.
-function payNativeGasForContractCallWithToken(
-    address sender,
-    string calldata destinationChain,
-    string calldata destinationAddress,
-    bytes calldata payload,
-    string calldata symbol,
-    uint256 amount,
-    address refundAddress
-) external payable;
-```
+See the [`AxelarGasService` interface](https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/ad37802dc6d62fff3ab589f2605f7a3e566977dd/contracts/interfaces/IAxelarGasService.sol) for alternative gas payment methods.
 
-### Methods arguments description
-The function names are prety self-explanatory. The following is true for the arguments:
+When passing in arguments for these methods:
 
-- For all functions:
-  - `sender` needs to match the address that calls `callContract` or `callContractWithToken` on the `AxelarGateway`. If the `AxelarGasReceiver` is called by the same contract that will call the Gateway, then simply specify `address(this)` as `sender`.
-- For `payNativeGasForContractCall`, the following need to match the arguments of a `contractCall` on the `AxelarGateway`:
-  - `destinationChain`
-  - `destinationAddress`
-  - `payload`
-- For `payNativeGasForContractCallWithToken`, the following need to match the arguments of a `contractCallWithToken` on the `AxelarGateway`:
-  - `destinationChain`
-  - `destinationAddress`
-  - `payload`
-  - `symbol`
-  - `amount`
-- For `payNativeGasForContractCall` and `payNativeGasForContractCallWithToken`, the amount of funds received is specified by `msg.value`.
-- For all functions, `refundAddress` is the address that will eventually be able to receive excess amount paid for gas.
+- `sender` must match the address that calls `callContract()` or `callContractWithToken()` on the `AxelarGateway`. If the `AxelarGasReceiver` is called by the same contract that will call the Gateway, specify `address(this)` as `sender`.
+- `refundAddress` is the address that will eventually be able to [receive the excess amount paid for gas](https://docs.axelar.dev/dev/general-message-passing/gas-services/refund).
```

Comment:
Should we clarify that refunds are made on the source chain?

Reply:
Updated here and in the intro